### PR TITLE
Add test case for invalid tz in timestamp literal

### LINKED ISSUE #8452: Invalid timezone information is ignored after `Z` in timestamp literals
### Describe the bug

Some timestamps that should error are actally parsed, leading to confusing cases when users make a mistake

For example this timestamp is invalid  `'2023-12-05T21:58:10.45ZZTOP'` but it is parsed as though it were `'2023-12-05T21:58:10.45Z'` (the trailing content is ignored)

### To Reproduce

using `datafusion-cli`, 

```sql
❯ select TIMESTAMP '2023-12-05T21:58:10.45ZZTOP';
+-------------------------------------+
| Utf8("2023-12-05T21:58:10.45ZZTOP") |
+-------------------------------------+
| 2023-12-05T21:58:10.450             |
+-------------------------------------+
1 row in set. Query took 0.001 seconds.
```

### Expected behavior

I expect the query to error, similarly to what happens if the `Z` is not present

```
❯ select TIMESTAMP '2023-12-05T21:58:10.45 hello';
Optimizer rule 'simplify_expressions' failed
caused by
Arrow error: Parser error: Invalid timezone "hello": 'hello' is not a valid timezone
```

This is consistent with postgres, which errors for such cases:

```sql
postgres=# select TIMESTAMP '2023-12-05T21:58:10.45ZZTOP';
ERROR:  invalid input syntax for type timestamp: "2023-12-05T21:58:10.45ZZTOP"
LINE 1: select TIMESTAMP '2023-12-05T21:58:10.45ZZTOP';
                         ^
postgres=# select timestamp '2023-12-05T21:58:10.45 hello';
ERROR:  invalid input syntax for type timestamp: "2023-12-05T21:58:10.45 hello"
LINE 1: select timestamp '2023-12-05T21:58:10.45 hello';
                         ^
postgres=# select timestamp '2023-12-05T21:58:10.45';
       timestamp
------------------------
 2023-12-05 21:58:10.45
(1 row)
```

### Additional context

Kudos to @reidkaufmann for finding this downstream 

This appears to be a bug in arrow: https://github.com/apache/arrow-rs/issues/5182

Once it is fixed in arrow, then we can write a test for it in DataFusion and close the issue

### PR DESCRIPTION
**Pull request by MohamedAbdeen21**
## Which issue does this PR close?

Closes #8452 .

## What changes are included in this PR?

A test case in `timestamps.slt`, and a test case in `datetime/to_timestamp.rs`.

## Are these changes tested?

Yes

## Are there any user-facing changes?

No

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thank you @MohamedAbdeen21 🙏 

Is there any chance you could add an slt test for this instead? 

Bascially add the query to 
https://github.com/apache/arrow-datafusion/blob/main/datafusion/sqllogictest/test_files/timestamps.slt

The instructions on how to run this type of test can be found here: https://github.com/apache/arrow-datafusion/tree/83d15e8b223e1af2c553ec54afa5c1ffac7c340e/datafusion/sqllogictest

**Comment by MohamedAbdeen21** [comment]
> Is there any chance you could add an slt test for this instead? 

Hi @alamb, the PR already contains a slt test. I wasn't sure where would be the best place, so I submitted both for review.

File: datafusion/sqllogictest/test_files/timestamps.slt [hunk 1]
```diff
@@ -465,6 +465,10 @@ SELECT TIME 'not a time' as time;
 query error Cannot cast string '24:01:02' to value of Time64\(Nanosecond\) type
 SELECT TIME '24:01:02' as time;
 
+# invalid timezone
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/timestamps.slt:468]: I am sorry -- I didn't see this initially.

File: datafusion/functions/src/datetime/to_timestamp.rs [hunk 2]
```diff
@@ -561,6 +561,31 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn to_timestamp_with_invalid_tz() -> Result<()> {
```
Comment (alamb) [inline datafusion/functions/src/datetime/to_timestamp.rs:565]: I think the slt coverage is sufficient, so this test is not necessary. However, I don't think that would stop this PR from merging.

**Review by alamb — APPROVED** [review]
Thank you so much @MohamedAbdeen21

**Comment by MohamedAbdeen21** [comment]
@alamb my pleasure, looking forward to contribute more

**Comment by alamb** [comment]
Thanks again @MohamedAbdeen21

**Pull request merged**